File: Python/Niveau 1/7 – Conditions avancées, opérateurs booléens/L'espion est démasqué !.py
personnes = int(input())

for i in range(personnes):
   size = int(input())
   age = int(input())
   weight = int(input())
   horse = int(input())
   hairB = int(input())
   good = 0
   
   if 182 >= size >= 178:
      good += 1
   if age >= 34:
      good += 1
   if weight < 70:
      good += 1
   if horse == 0:
      good += 1
   if hairB == 1:
      good += 1

   if good == 5:
      print("Très probable")
   elif good >= 3:
      print("Probable")
   elif good == 0:
      print("Impossible")
   else:
      print("Peu probable")

File: Pascal/Niveau 1/7 – Conditions avancées, opérateurs booléens/L'espion est démasqué !.pas
program Solution;
var
   nbPersonnes, nbCriteres: LongInt;
   size, age, weight, horse, brownHair: LongInt;
   loop: LongInt;
begin
   read(nbPersonnes);
   for loop := 1 to nbPersonnes do
   begin
      nbCriteres := 0;
      read(size);
      if (178 <= size) and (size <= 182) then
      begin
         nbCriteres := nbCriteres + 1;
      end;
      read(age);
      if age >= 34 then
      begin
         nbCriteres := nbCriteres + 1;
      end;
      read(weight);
      if weight < 70 then
      begin
         nbCriteres := nbCriteres + 1;
      end;
      read(horse);
      if horse = 0 then
      begin
         nbCriteres := nbCriteres + 1;
      end;
      read(brownHair);
      if brownHair = 1 then
      begin
         nbCriteres := nbCriteres + 1;
      end;   
      
      if nbCriteres = 0 then
      begin
         writeln('Impossible');
      end
      else if nbCriteres = 5 then
      begin
         writeln('Très probable');
      end
      else if nbCriteres >= 3 then
      begin
         writeln('Probable');
      end
      else
      begin
         writeln('Peu probable');
      end 
   end;
end.